app/src/main/java/io/apicurio/registry/util/ArtifactTypeUtil.java

```diff
@@ -84,20 +113,47 @@ public static ArtifactType discoverType(ContentHandle content, String contentTyp
         } catch (Exception e) {
             // Apparently it's not JSON.
         }
-        
+
         // Try protobuf (only if we haven't already)
         if (!triedProto) {
             ArtifactType type = tryProto(content);
             if (type != null) {
                 return type;
             }
         }
-        
+
         // Try GraphQL (SDL)
         if (tryGraphQL(content)) {
             return ArtifactType.GRAPHQL;
         }
-        
+
+        // Try the various XML formatted types
+        try (InputStream stream = content.stream()) {
+            Document xmlDocument = threadLocaldocBuilder.get().parse(stream);
+            Element root = xmlDocument.getDocumentElement();
+            String ns = root.getNamespaceURI();
+
+            if (ns == null) {
+                // XML
+                return ArtifactType.XML;
+            }
+
+            // XSD
+            if (ns.contains("www.w3.org/2001/XMLSchema")) {
+                return ArtifactType.XSD;
+            }
```

Comment:
We need to be more specific here - don't use contains.  Use `equals` and specifically check for the XSD namespace:  `http://www.w3.org/2001/XMLSchema`

Reply:
I do not believe there are multiple versions of XML Schema that we have to check for.

Reply:
👍